disref/process.py

```diff
@@ -100,23 +106,72 @@ def create_reference(self, resource, block=True):
         :returns: The created Reference object
         """
 
-        self.client.hset(self.registry_key, resource, 1)
+        self.add_to_registry(resource)
         return Reference(self, resource, block)
 
+    def add_to_registry(self, resource, registry_key=None):
```

Comment:
We should indicate in `add_to_registry` and `remove_from_registry` documentation that they are not threadsafe, as well as any method that calls them up the stack.